homeassistant/components/comelit/coordinator.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Comelit."""
+import asyncio
+from collections.abc import Callable
+from datetime import timedelta
+
+from aiocomelit import (
+    ComeliteSerialBridgeAPi,
+    ComelitSerialBridgeObject,
+    ComelitVedoObject,
+)
+import aiohttp
+
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import _LOGGER, DOMAIN
+
+
+class ComelitSerialBridge(DataUpdateCoordinator):
+    """Queries Comelit Serial Bridge."""
+
+    def __init__(self, host: str, pin: int, hass: HomeAssistant) -> None:
+        """Initialize the scanner."""
+
+        self._host = host
+        self._pin = pin
+        self._devices: list[ComelitSerialBridgeObject] = []
+        self._alarm: list[ComelitVedoObject] = []
+        self._on_close: list[Callable] = []
+
+        self.api = ComeliteSerialBridgeAPi(host, pin)
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{DOMAIN}-{host}-coordinator",
+            update_interval=timedelta(seconds=5),
+        )
+
+    @callback
+    def async_listener(self) -> None:
+        """Fetch data from the router."""
+
+        asyncio.run_coroutine_threadsafe(self._async_update_data(), self.hass.loop)
```

Comment:
I don't understand what's happening here.
A coordinator will call its listeners after a refresh, i.e. after `_async_update_data` has been called. This listener will create a task which will immediately do another refresh as I understand it?

Reply:
Removed